torch/distributed/_tensor/placement_types.py

```diff
@@ -302,21 +304,22 @@ def __str__(self) -> str:
         """
         return "R"
 
-    def _replicate_tensor(
-        self, tensor: torch.Tensor, mesh: DeviceMesh, mesh_dim: int
-    ) -> torch.Tensor:
+    def _replicate_tensor(self, tensor: torch.Tensor, mesh: DeviceMesh) -> torch.Tensor:
         """
-        Replicate (broadcast) a torch.Tensor on a mesh dimension (use
-        the first coordinate on the mesh dimension as source of truth)
+        Replicate a torch.Tensor by copying it to the local device.
         """
         my_coordinate = mesh.get_coordinate()
         if my_coordinate is None:
             # if rank is not part of mesh, we simply return an empty tensor
-            return tensor.new_empty(0, requires_grad=tensor.requires_grad)
+            return tensor.new_empty(
+                0, device=mesh.device_type, requires_grad=tensor.requires_grad
+            )
 
-        tensor = tensor.contiguous()
-        mesh.broadcast(tensor, mesh_dim=mesh_dim)
-        return tensor
+        output = torch.empty_like(tensor, device=mesh.device_type)
+        if not tensor.is_meta:
+            output.copy_(tensor)
```

Comment:
Similarly, we can't simply copy the corresponding replica as that would result in different rank have different data, even though the intention of `replicate_tensor` is to replicate the data.

Random seed is important to ensure model correctness so we would have to do a broadcast in this case.

Reply:
yes, thanks. i will revert it